Makefile

```diff
@@ -1,119 +1,85 @@
-USE_FRAMEBUFFER := 1
-SMALL_PAGES := 0
-CFLAGS := -std=gnu99 \
-        -ffreestanding \
-        -O2 \
-        -Wall \
-        -Wextra \
-        -I src/include \
-        -I src/include/base \
-        -I src/include/kernel \
-        -I src/include/kernel/mem \
-        -I src/include/kernel/x86_64 \
-        -I src/include/kernel/hardware \
-        -I src/include/kernel/scheduling \
-        -I src/include/libc \
-        -I src/include/fs \
-        -I src/include/drivers/fs \
-        -I src/include/sys \
-        -mno-red-zone \
-        -mno-sse \
-        -mcmodel=large \
-        -DUSE_FRAMEBUFFER=$(USE_FRAMEBUFFER) \
-        -DSMALL_PAGES=$(SMALL_PAGES)
-
-TESTFLAGS := -std=gnu99 \
-        -I tests/include \
-        -I src/include \
-        -I src/include/base \
-        -I src/include/kernel/mem \
-        -I src/include/fs \
-        -I src/include/drivers/fs \
-        -I src/include/kernel \
-        -I src/include/kernel/x86_64 \
-        -I src/include/sys \
-        -DSMALL_PAGES=$(SMALL_PAGES) \
-        -D_TEST_=1
-
-TEST_FOLDER := tests
-C_DEBUG_FLAGS := -g \
-				-DDEBUG=1
-NASM_DEBUG_FLAGS := -g \
+include build/Config.mk
+include build/Common.mk
+
+# arch dependant
+ARCH_PREFIX := x86_64-elf
+ASM_COMPILER := nasm
+QEMU_SYSTEM := qemu-system-x86_64
+
+ASM_DEBUG_FLAGS := -g \
 					-F dwarf
-NASMFLAGS := -f elf64 \
+ASM_FLAGS := -f elf64 \
 		-D USE_FRAMEBUFFER=$(USE_FRAMEBUFFER) \
```

Comment:
I know it's only two variables: but since you're using the same defines for both asm and C, maybe create an intermediate variable and then just append that:
```
DEF_FLAGS =  -DUSE_FRAMEBUFFER=$(USE_FRAMEBUFFER) -DSMALL_PAGES=$(SMALL_PAGES)|

ASM_FLAGS = -f elf64
ASM_FLAGS += $(DEF_FLAGS)

C_FLAGS += $(DEF_FLAGS)
```

Reply:
obviously put the C_FLAGS line wherever is appropriate for you